src/core/webview/ClineProvider.ts

```diff
@@ -795,39 +795,100 @@ export class ClineProvider extends EventEmitter<ClineProviderEvents> implements
 		await this.postStateToWebview()
 	}
 
+	// Provider Profile Management
+
+	getProviderProfileEntries(): ProviderSettingsEntry[] {
+		return this.contextProxy.getValues().listApiConfigMeta || []
+	}
+
+	getProviderProfileEntry(name: string): ProviderSettingsEntry | undefined {
+		return this.getProviderProfileEntries().find((profile) => profile.name === name)
+	}
+
+	public hasProviderProfileEntry(name: string): boolean {
+		return !!this.getProviderProfileEntry(name)
+	}
+
+	async upsertProviderProfile(name: string, providerSettings: ProviderSettings): Promise<string | undefined> {
```

Comment:
I renamed `upsertApiConfiguration` to `upsertProviderProfile` and got rid of `updateApiConfiguration` since it was only being called by the `upsertProviderProfile` and `activateProviderProfile`, and was something internally redundant with those.

Reply:
Update: I had Roo go ahead and make this change :)